src/Func.cpp

```diff
@@ -1010,6 +1010,14 @@ void Stage::split(const string &old, const string &outer, const string &inner, c
 
     definition.schedule().touched() = true;
 
+    if (var_name_match(outer, inner)) {
```

Comment:
var_name_match isn't right here. It's for comparing thing in the dims list (which are qualified) to Var names (which are unqualified). Just do user_assert(outer != inner) << ...